server/src/main/java/io/pravega/schemaregistry/service/SchemaRegistryService.java

```diff
@@ -264,7 +264,7 @@ public SchemaRegistryService(SchemaStore store, ScheduledExecutorService executo
         Preconditions.checkArgument(group != null);
         Preconditions.checkArgument(schemaInfo != null);
         log.debug("addSchema called for group {} {}. schema {}", namespace, group, schemaInfo.getType());
-        SchemaInfo schema = normalizeSchemaBinary(schemaInfo);
+        SchemaInfo schema = normalizeSchemaBinary(schemaInfo, group, namespace);
```

Comment:
As discussed, if namespace passed by user is null or empty, we should replace it with default namespace.